store/etcd3_store.go

```diff
@@ -0,0 +1,204 @@
+package store
+
+import (
+	"context"
+
+	"fmt"
+	"strings"
+
+	"github.com/cenkalti/backoff"
+	"github.com/coreos/etcd/client"
+	"github.com/coreos/etcd/clientv3"
+	"github.com/golang/protobuf/proto"
+	log "github.com/sirupsen/logrus"
+	"github.com/sky-uk/merlin/types"
+)
+
+type etcd3store struct {
+	client *clientv3.Client
+	prefix string
+}
+
+// NewEtcd3 returns a Store implementation using an etcd3 backing store.
+func NewEtcd3(endpoints []string, prefix string) (Store, error) {
+	cfg := clientv3.Config{
+		Endpoints: endpoints,
```

Comment:
I don't think there is a direct equivalent as the etcd3 is using grpc underneath, whereas the v2 client was using HTTP calls. We can set a dial timeout, but that is only for the initial connection. Perhaps this per-request timeouts are to be managed by the context in v3? We could possibly set a timeout there to replicate this existing behaviour on V2?

Reply:
I think we can set the dial timeout to avoid waiting for a long time, and let the overall timeout be handled by the propagated context coming from the client.